roachtest: node-postgres failed

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.node-postgres [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/20924562?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/20924562?buildTab=artifacts#/node-postgres) on master @ [f84948abd7db42f8f1030cbe1f9434e3045755b8](https://github.com/cockroachdb/cockroach/commits/f84948abd7db42f8f1030cbe1f9434e3045755b8):

Failed with:

```
(orm_helpers.go:225).summarizeFailed: 
Tests run on Cockroach v26.1.0-alpha.2-dev-f84948abd7db42f8f1030cbe1f9434e3045755b8
Tests run against node-postgres N/A
171 Total Tests Run
168 tests passed
3 tests failed
2 tests skipped
2 tests ignored
0 tests passed unexpectedly
3 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: idle timeout => does not time out clients which are used - unknown (unexpected)
--- FAIL: lifetime timeout => connection lifetime should expire and remove the client after the client is done working - unknown (unexpected)
--- FAIL: idle timeout => can remove idle clients and recreate them - unknown (unexpected)
For a full summary look at the node-postgres artifacts 
An updated blocklist (nodePostgresBlockList) is available in the artifacts' node-postgres log
test artifacts and logs in: /artifacts/node-postgres/run_1
```
Cluster Node to Ip Mapping:
| Node | Public IP | Private IP |
| --- | --- | --- |
| teamcity-20924562-1767509199-186-n1cpu4-0001 | 20.55.77.151 | 10.1.0.110 |

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=azure</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=default</code>
 - <code>metamorphicWriteBuffering=true</code>
 - <code>runtimeAssertionsBuild=true</code>
 - <code>ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

_Grafana is not yet available for azure clusters_</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*node-postgres.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-58363

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.node-postgres [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/21078675?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/21078675?buildTab=artifacts#/node-postgres) on master @ [2a299a81cf102d216e943440bcc027720f939f9d](https://github.com/cockroachdb/cockroach/commits/2a299a81cf102d216e943440bcc027720f939f9d):

Failed with:

```
(orm_helpers.go:225).summarizeFailed: 
Tests run on Cockroach v26.2.0-alpha.00000000-dev-2a299a81cf102d216e943440bcc027720f939f9d
Tests run against node-postgres N/A
172 Total Tests Run
166 tests passed
6 tests failed
2 tests skipped
2 tests ignored
0 tests passed unexpectedly
6 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: lifetime timeout => connection lifetime should expire and remove the client - unknown (unexpected)
--- FAIL: idle timeout => can remove idle clients and recreate them - unknown (unexpected)
--- FAIL: pool => with promises => recovers from query errors - unknown (unexpected)
--- FAIL: lifetime timeout => connection lifetime should expire and remove the client after the client is done working - unknown (unexpected)
--- FAIL: lifetime timeout => can remove expired clients and recreate them - unknown (unexpected)
--- FAIL: pool => with promises => supports just running queries - unknown (unexpected)
For a full summary look at the node-postgres artifacts 
An updated blocklist (nodePostgresBlockList) is available in the artifacts' node-postgres log
test artifacts and logs in: /artifacts/node-postgres/run_1
```
Cluster Node to Ip Mapping:
| Node | Public IP | Private IP |
| --- | --- | --- |
| teamcity-21078675-1770447423-115-n1cpu4-0001 | 20.55.98.98 | 10.1.0.115 |

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=azure</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>diskCount=0</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=expiration</code>
 - <code>metamorphicWriteBuffering=true</code>
 - <code>runtimeAssertionsBuild=true</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

_Grafana is not yet available for azure clusters_</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*node-postgres.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.node-postgres [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/21108031?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/21108031?buildTab=artifacts#/node-postgres) on master @ [817f6e41bd57ff4c2ea545d91bac5aaad2637788](https://github.com/cockroachdb/cockroach/commits/817f6e41bd57ff4c2ea545d91bac5aaad2637788):

Failed with:

```
(orm_helpers.go:225).summarizeFailed: 
Tests run on Cockroach v26.2.0-alpha.1-dev-817f6e41bd57ff4c2ea545d91bac5aaad2637788
Tests run against node-postgres N/A
171 Total Tests Run
168 tests passed
3 tests failed
2 tests skipped
2 tests ignored
0 tests passed unexpectedly
3 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: lifetime timeout => connection lifetime should expire and remove the client after the client is done working - unknown (unexpected)
--- FAIL: pool => with promises => recovers from query errors - unknown (unexpected)
--- FAIL: lifetime timeout => connection lifetime should expire and remove the client - unknown (unexpected)
For a full summary look at the node-postgres artifacts 
An updated blocklist (nodePostgresBlockList) is available in the artifacts' node-postgres log
test artifacts and logs in: /artifacts/node-postgres/run_1
```
Cluster Node to Ip Mapping:
| Node | Public IP | Private IP |
| --- | --- | --- |
| teamcity-21108031-1771052228-103-n1cpu4-0001 | 20.127.160.222 | 10.1.0.11 |

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=azure</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>diskCount=0</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=true</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

_Grafana is not yet available for azure clusters_</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*node-postgres.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.node-postgres [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/21108902?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/21108902?buildTab=artifacts#/node-postgres) on master @ [817f6e41bd57ff4c2ea545d91bac5aaad2637788](https://github.com/cockroachdb/cockroach/commits/817f6e41bd57ff4c2ea545d91bac5aaad2637788):

Failed with:

```
(orm_helpers.go:225).summarizeFailed: 
Tests run on Cockroach v26.2.0-alpha.1-dev-817f6e41bd57ff4c2ea545d91bac5aaad2637788
Tests run against node-postgres N/A
171 Total Tests Run
170 tests passed
1 test failed
2 tests skipped
2 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: Bring your own promise => uses supplied promise for operations - unknown (unexpected)
For a full summary look at the node-postgres artifacts 
An updated blocklist (nodePostgresBlockList) is available in the artifacts' node-postgres log
test artifacts and logs in: /artifacts/node-postgres/run_1
```
Cluster Node to Ip Mapping:
| Node | Public IP | Private IP |
| --- | --- | --- |
| teamcity-21108902-1771138524-96-n1cpu4-0001 | 20.55.91.98 | 10.1.0.117 |

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=azure</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>diskCount=0</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=epoch</code>
 - <code>metamorphicWriteBuffering=true</code>
 - <code>runtimeAssertionsBuild=false</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

_Grafana is not yet available for azure clusters_</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*node-postgres.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.node-postgres [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/21139178?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/21139178?buildTab=artifacts#/node-postgres) on master @ [77b654b08c6f6717868d3b279ac028e4d87aaf17](https://github.com/cockroachdb/cockroach/commits/77b654b08c6f6717868d3b279ac028e4d87aaf17):

Failed with:

```
(orm_helpers.go:225).summarizeFailed: 
Tests run on Cockroach v26.2.0-alpha.1-dev-77b654b08c6f6717868d3b279ac028e4d87aaf17
Tests run against node-postgres N/A
171 Total Tests Run
169 tests passed
2 tests failed
2 tests skipped
2 tests ignored
0 tests passed unexpectedly
2 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: lifetime timeout => connection lifetime should expire and remove the client after the client is done working - unknown (unexpected)
--- FAIL: pool error handling => pool with lots of errors => continues to work and provide new clients - unknown (unexpected)
For a full summary look at the node-postgres artifacts 
An updated blocklist (nodePostgresBlockList) is available in the artifacts' node-postgres log
test artifacts and logs in: /artifacts/node-postgres/run_1
```
Cluster Node to Ip Mapping:
| Node | Public IP | Private IP |
| --- | --- | --- |
| teamcity-21139178-1771657070-102-n1cpu4-0001 | 48.217.234.57 | 10.1.0.104 |

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=azure</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>diskCount=0</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=default</code>
 - <code>runtimeAssertionsBuild=false</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

_Grafana is not yet available for azure clusters_</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*node-postgres.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.node-postgres [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/21140124?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/21140124?buildTab=artifacts#/node-postgres) on master @ [66d2276bce4edaba5d105cd21f59f3a301cd3821](https://github.com/cockroachdb/cockroach/commits/66d2276bce4edaba5d105cd21f59f3a301cd3821):

Failed with:

```
(orm_helpers.go:225).summarizeFailed: 
Tests run on Cockroach v26.2.0-alpha.1-dev-66d2276bce4edaba5d105cd21f59f3a301cd3821
Tests run against node-postgres N/A
171 Total Tests Run
168 tests passed
3 tests failed
2 tests skipped
2 tests ignored
0 tests passed unexpectedly
3 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: connection timeout => continues processing after a connection failure - unknown (unexpected)
--- FAIL: pool => with promises => connects, queries, and disconnects - unknown (unexpected)
--- FAIL: pool => with callbacks => does not create promises when querying - unknown (unexpected)
For a full summary look at the node-postgres artifacts 
An updated blocklist (nodePostgresBlockList) is available in the artifacts' node-postgres log
test artifacts and logs in: /artifacts/node-postgres/run_1
```
Cluster Node to Ip Mapping:
| Node | Public IP | Private IP |
| --- | --- | --- |
| teamcity-21140124-1771743311-102-n1cpu4-0001 | 20.127.185.54 | 10.1.0.162 |

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=azure</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>diskCount=0</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=expiration</code>
 - <code>runtimeAssertionsBuild=false</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

_Grafana is not yet available for azure clusters_</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*node-postgres.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/cmd/roachtest/tests/nodejs_postgres_blocklist.go
+++ b/pkg/cmd/roachtest/tests/nodejs_postgres_blocklist.go
@@ -15,7 +15,13 @@ package tests
 var nodePostgresBlockList = blocklist{}
 
 var nodePostgresIgnoreList = blocklist{
-	"events => emits acquire every time a client is acquired":             "flaky",
-	"pool => with callbacks => removes client if it errors in background": "152728",
-	"pool size of 1 => can only send 1 query at a time":                   "152728",
+	"connection timeout => releases newly connected clients if the queued already timed out":                       "flaky",
+	"events => emits acquire every time a client is acquired":                                                      "flaky",
+	"lifetime timeout => can remove expired clients and recreate them":                                             "flaky",
+	"lifetime timeout => connection lifetime should expire and remove the client after the client is done working": "flaky",
+	"pool ending => ends with clients":                                                                             "flaky",
+	"pool error handling => pool with lots of errors => continues to work and provide new clients":                 "flaky",
+	"pool => with callbacks => removes client if it errors in background":                                          "152728",
+	"pool => with callbacks => works totally unconfigured":                                                         "flaky",
+	"pool size of 1 => can only send 1 query at a time":                                                            "152728",
 }

